src/OrchardCore/OrchardCore.Setup.Abstractions/Events/ITenantSetupHandler.cs

```diff
@@ -0,0 +1,26 @@
+using System.Threading.Tasks;
+using OrchardCore.Setup.Services;
+
+namespace OrchardCore.Setup.Events;
+
+public interface ITenantSetupHandler
+{
+    /// <summary>
+    /// Called during the process of setting up a new tenant.
+    /// </summary>
+    /// <param name="context"></param>
+    /// <returns></returns>
+    Task SetupAsync(SetupContext context);
+
+    /// <summary>
+    /// Called when a new tenant is successfully setup.
+    /// </summary>
+    /// <returns></returns>
+    Task SucceededAsync();
+
+    /// <summary>
+    /// Called when a tenant fails to setup.
+    /// </summary>
+    /// <returns></returns>
+    Task FailedAsync(SetupContext context);
```

Comment:
I would define these methods in the same order than in the base class.

And in this order `SetupAsync()`, `FailedAsync()` and `SucceededAsync()`.